bser.c

```diff
@@ -353,7 +380,16 @@ int w_bser_dump(const bser_ctx_t* ctx, json_t *json, void *data)
     case JSON_INTEGER:
       return bser_int(ctx, json_integer_value(json), data);
     case JSON_STRING:
-      return bser_bytestring(ctx, json_string_value(json), data);
+      wstr = json_to_w_string(json);
+      switch (wstr->type) {
```

Comment:
Add a comment here explaining that `bser_utf8string` is doing to do the version/capabilities check